internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -440,7 +440,9 @@ func (r *Reconciler) reconcileCluster(ctx context.Context, s *scope.Scope) error
 
 	// Check differences between current and desired state, and eventually patch the current object.
 	patchHelper, err := r.patchHelperFactory(s.Current.Cluster, s.Desired.Cluster, structuredmerge.IgnorePaths{
-		{"spec", "controlPlaneEndpoint"}, // this is a well known field that is managed by the Cluster controller, topology should not express opinions on it.
+		// This is a well known field that is managed by the Cluster controller, topology should not express opinions on it.
+		// It needs to be added here because it is not a pointer at the struct and would otherwise get defaulted by the topology controller.
+		{"spec", "controlPlaneEndpoint"},
```

Comment:
This means we still have to add ignorePaths for all the non-pointer fields which have a similar issue?

I also prefer this variant, but that we have to add spec.controlPlaneEndpoint makes me wonder if we should use allowPaths additionally to make it future proof

Reply:
I just think this might end badly if we add new fields to the Cluster and don't remember this issue, which is why I would prefer having allowPaths here.

We could keep it simple and just export allowPaths and use them instead of hard-coded ones if the option is set.

Reply:
> At the end this should result in failing tests.

This assumes that we have e2e test coverage for this field

Reply:
Missed that we have care about some labels as well :joy:

Reply:
I adjusted the helper.

I added an additional option which allows pass a more specific list instead of `{"spec"}` to handle the cluster use case.